stratum/hal/lib/tai/test/tai_manager_test.cc

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "tai.h"
+
+#include "stratum/hal/lib/tai/hostinterface.h"
+#include "stratum/hal/lib/tai/module.h"
+#include "stratum/hal/lib/tai/networkinterface.h"
+#include "stratum/hal/lib/tai/taiadapterhost.h"
+#include "stratum/hal/lib/tai/taimanager.h"
+
+namespace stratum {
+namespace hal {
+namespace tai {
+
+class TAIManagerTest : public ::testing::Test {
+ public:
+  static void SetUpTestCase() { manager_ = TAIManager::Instance(); }
+  static void TearDownTestCase() {
+    manager_ = nullptr;
+    TAIManager::Delete();
+  }
+
+  static TAIManager *manager_;
```

Comment:
Don't make the manager a static raw pointer. Current practice is using a `protected std::unique_ptr<TAIManager> manager_;`